src/google-classroom-extractor/google_classroom_extractor/api/courses.py

```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import logging
+from typing import List, Dict
+import pandas as pd
+from .api_caller import call_api
+
+
+def request_courses(resource) -> List[Dict[str, str]]:
+    # type: ignore
+    return call_api(
+        resource.courses().list,
```

Comment:
A quick-and-dirty test to confirm something...

```python
class Test_annotations:
    def test_does_type_checking_stop_none(self):

        def my_method_with_str(arg: str):
            print("my_method_with_str: ", str)

        def my_method_with_str_2(arg: str):
            assert arg is not None

        my_method_with_str(None)
        my_method_with_str_2(None)
```

Confirmed: type annotation won't keep you from passing in a `None`. This won't surprise any of us. Just good to see it.

So, what happens if `resource` is `None`? An `AttributeError` would be raised on the attempt to call `courses()`. This is where I would prefer to see the function start with `assert resource is not None`.